### PR TITLE
raftstore: clean temporary file when startup

### PR DESCRIPTION
**Pull request by BusyJay**
If temporary file is not cleaned up and there is no new data written in, the snapshot may not be able to be generated or received again.

@siddontang @hhkbp2 PTAL

### CODE REVIEW

File: src/raftstore/store/snap.rs [hunk 1]
```diff
@@ -15,6 +15,8 @@ use kvproto::raft_serverpb::RaftSnapshotData;
 use raftstore::store::Msg;
 use util::transport::SendCh;
 
+const TMP_FILE_SUFFIX: &'static str = ".tmp";
```
Comment (ngaut) [inline src/raftstore/store/snap.rs]: s/tmp/tikv-tmp/
In case some applications using "tmp" as suffix.

Comment (BusyJay) [inline src/raftstore/store/snap.rs]: It's ok, the directory is created by TiKV and only used for TiKV.

File: src/raftstore/store/snap.rs [hunk 2]
```diff
@@ -371,3 +383,56 @@ pub type SnapManager = Arc<RwLock<SnapManagerCore>>;
...
+        mgr.wl().init().unwrap();
+        assert!(p.exists());
+
+        // if target is a file, an error should be returned.
+        let path2 = path.path().to_str().unwrap().to_owned() + "/snap2";
+        File::create(&path2).unwrap();
+        mgr = new_snap_mgr(path2, None);
+        assert!(mgr.wl().init().is_err());
+
+        // if temporary files exist, they should deleted.
```
Comment (siddontang) [inline src/raftstore/store/snap.rs]: be deleted

**Comment by siddontang** [comment]
Rest LGTM

**Comment by BusyJay** [comment]
PTAL

**Comment by siddontang** [comment]
LGTM
PTAL @hhkbp2

File: src/raftstore/store/snap.rs [hunk 3]
```diff
@@ -371,3 +383,56 @@ pub type SnapManager = Arc<RwLock<SnapManagerCore>>;
...
+    use tempdir::TempDir;
+
+    use util::HandyRwLock;
+    use super::*;
+
+    #[test]
+    fn test_snap_mgr() {
+        let path = TempDir::new("test-snap-mgr").unwrap();
+
+        // mgr should create directory when not exist.
```
Comment (hhkbp2) [inline src/raftstore/store/snap.rs]: Revise this comment to 
"`mgr` should create the specified directory when it does not exist."

**Comment by hhkbp2** [comment]
Rest LGTM

**Comment by ngaut** [comment]
LGTM

**Pull request merged**